supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -415,36 +477,83 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
+          startTime <- responseLogs.head.start_time
+          vmCost <- responseLogs.head.metadata.flatMap(_.get("vm_price_per_hour_usd"))
+        } yield (startTime, vmCost)
+
+        fetchedCostData match {
+          case Some(data) =>
+            val costData = Option(TesVmCostData(data._1, data._2))
+            val state = response.state
+            val metadata = Map(
+              CallMetadataKeys.TaskStartTime -> data._1,
+              CallMetadataKeys.VmCostUsd -> data._2
+            )
+            tellMetadata(metadata)
+            getTesStatus(state, costData, handle.pendingJob.jobId)
+          case None => getTesStatus(response.state, Option.empty, handle.pendingJob.jobId)
+        }
+      }
+    } else {
+      makeRequest[MinimalTaskView](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=MINIMAL")) map {
+        response =>
+          val state = response.state
+          getTesStatus(Option(state), Option.empty, handle.pendingJob.jobId)
+      }
+    }
+
+  private def getTesStatus(state: Option[String], withCostData: Option[TesVmCostData], jobId: String): TesRunStatus = {
+    var data: Option[TesVmCostData] = Option.empty
+    if (withCostData.isDefined) {
+      data = withCostData
+      for {
+        costData <- withCostData
+        vmCost = costData.vmCost
+      } yield vmCost
+    }
+    state match {
+      case s if s.contains("COMPLETE") =>
+        jobLogger.info(s"Job ${jobId} is complete")
+        Complete(data)
 
-          case s if s.contains("CANCELED") =>
-            jobLogger.info(s"Job ${handle.pendingJob.jobId} was canceled")
-            Cancelled
+      case s if s.contains("CANCELED") =>
+        jobLogger.info(s"Job ${jobId} was canceled")
+        Cancelled(data)
 
-          case s if s.contains("EXECUTOR_ERROR") =>
-            jobLogger.info(s"TES reported a failure for Job ${handle.pendingJob.jobId}: '$s'")
-            Failed()
+      case s if s.contains("EXECUTOR_ERROR") =>
+        jobLogger.info(s"TES reported a failure for Job ${jobId}: '$s'")
+        Failed()
 
-          case s if s.contains("SYSTEM_ERROR") =>
-            jobLogger.info(s"TES reported an error for Job ${handle.pendingJob.jobId}: '$s'")
-            Error()
+      case s if s.contains("SYSTEM_ERROR") =>
+        jobLogger.info(s"TES reported an error for Job ${jobId}: '$s'")
+        Error()
 
-          case _ => Running
-        }
+      case _ => Running(data)
     }
+  }
 
   private def getErrorLogs(handle: StandardAsyncPendingExecutionHandle): Future[Seq[String]] =
     makeRequest[Task](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=FULL")) map { response =>
       response.logs.flatMap(_.lastOption).flatMap(_.system_logs).getOrElse(Seq.empty[String])
     }
 
+  private def getTaskLogs(handle: StandardAsyncPendingExecutionHandle): Future[TaskLog] =
+    makeRequest[Task](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=FULL")) map { response =>
```

Comment:
We're now repeating this line in a few places, would be nice to factor it and its sibling into `fetchFullTesTaskStatus` and `fetchMinimalTesTaskStatus`.